src/network/networkd-network.h

```diff
@@ -191,6 +191,11 @@ struct Network {
         uint32_t br_vid_bitmap[BRIDGE_VLAN_BITMAP_LEN];
         uint32_t br_untagged_bitmap[BRIDGE_VLAN_BITMAP_LEN];
 
+        /* CAN support */
+        uint32_t can_bitrate;
+        uint32_t can_sample_point;
```

Comment:
hmm, config_parse_permille() currently expects to write to an "int" location, you define this here as "uint32_t"... Of course on all systems we care about the difference doesn't matter, given that "int" is 32bit, and we know that the parsed value is between 0…1000. However, I think we should be more careful here, and not access "uint32" data fields through "int" pointers.

I'd probably settle for using the "unsigned" type here. i.e. change config_parse_permille() to cast the return value of parse_permille() to an unsigned after handling the errors it might return, and then assigning it to an unsigned pointer. And hence change the field in the Network structure to an unsigned, too. Why "unsigned"? Well, it's relatively generic, and covers the 0…1000 range pretty closely, which is all we need here...

Reply:
`parse_permille()` does return an `int`, with `-errno` in case of error and a value from `0...1000` otherwise. `config_parse_permille()` will ignore the any errors (as most `config_parse_*()` do), so I'll make `struct Network::can_cample_point` an int. In `networkd-link.c` `link_set_can()` I'll trust that `network->can_sample_point` is `0...1000` and convert it without further checks into an `uint32_t`. Is this OK?

Reply:
Well, all other variables we store percent values in so far used "unsigned". Hence I'd really pick the same type for permille too, even if it means casting from int to unsigned after handling the parse error.

I mean, this doesn't matter much, it's mostly a question of internal consistency, and least surprises.

I mean, if you store a percent/permille value in an `unsigned`, then any value you could assign it to makes sense. if you store it in a simple `int`, then things are a bit weird because negative perent/permille are really weird in almost all cases...

But dunno, this is not a biggie

Reply:
Using `unsigned` now.